### PR TITLE
⬆️ UPDATE: Sync with mdformat 0.6.0 (tree rendering)

### PR DESCRIPTION
**Pull request by hukkinj1**
~~Sketching out how the plugin would look like after adapting to the changes in git+https://github.com/hukkinj1/mdformat.git@tree-structure~~

Edit: Ready for review!

### CODE REVIEW

**Comment by chrisjsewell** [comment]
@hukkinj1, to confirm, does this still pass all the tests when using https://github.com/executablebooks/mdformat/pull/132

**Comment by hukkinj1** [comment]
Yes all tests pass if you `pip install git+https://github.com/hukkinj1/mdformat.git@tree-structure` instead of the current `mdformat` PyPI release.

**Comment by hukkinj1** [comment]
@chrisjsewell This now uses mdformat from PyPI so should be ready for review

**Comment by chrisjsewell** [comment]
are you still wanting to merge this? (what with all the changes to `SyntaxTree`)

**Comment by hukkinj1** [comment]
Yup this is ready for review (and merge from my perspective)! The last PR that added `walk` method is something that can improve this, but by no means mandatory. This can adapt later (only after both markdown-it-py and mdformat get their next release).

**Review by chrisjsewell — APPROVED** [review]

File: mdformat_tables/plugin.py [hunk 1]
```diff
@@ -12,26 +12,18 @@ def update_mdit(mdit: MarkdownIt) -> None:
...
-                    options,
-                    env,
-                    finalize=False,
-                )
-                .replace(MARKERS.BLOCK_SEPARATOR, "")
-                .rstrip()
-            )
-    return rows
+    """Convert nodes in each cell to strings."""
+    return [[cell.render(renderer_funcs, options, env) for cell in row] for row in rows]
```
Comment (chrisjsewell) [inline mdformat_tables/plugin.py:21]: I assume there is no longer the issue that inline text needs to be in a paragraph to be render correctly

Comment (hukkinj1) [reply mdformat_tables/plugin.py:21]: Hmm, not sure but I would my gut says there is no such Markdown text that would parse into inline text outside a paragraph. If there is a case we should add a test.

Also the code that adds a surrounding paragraph was buggy, (line 23 in the red diff is dead code) yet I never bumped into any issues, so I assume that is not needed.

Comment (chrisjsewell) [reply mdformat_tables/plugin.py:21]: > my gut says there is no such Markdown text that would parse into inline text outside a paragraph.

Thats exactly how text in table cells work; they are not enclosed in a paragraph, just `th` or `td` tags

Comment (hukkinj1) [reply mdformat_tables/plugin.py:21]: Ah ok, but tests pass so :man_shrugging: 

Seems to work! :smile:

Comment (hukkinj1) [reply mdformat_tables/plugin.py:21]: Oh btw, this code passes all GFM spec tests in https://github.com/hukkinj1/mdformat-gfm/pull/2

So I'm very confident.

Comment (chrisjsewell) [reply mdformat_tables/plugin.py:21]: Yeh if the tests pass then should be fine 👍 it was just to note this change in behaviour

**Pull request merged**